src/topology/metric_space/hausdorff_distance.lean

```diff
@@ -743,4 +743,75 @@ by simp [(Hausdorff_edist_zero_iff_eq_of_closed hs ht).symm, Hausdorff_dist,
          ennreal.to_real_eq_zero_iff, fin]
 
 end --section
+
+section thickening
+
+variables {α : Type u} [pseudo_emetric_space α]
+
+open emetric
+
+/-- The (open) `δ`-thickening `thickening δ E` of a subset `E` in a pseudo emetric space consists
+of those points that are at distance less than `δ` from some point of `E`. -/
+def thickening (δ : ℝ) (E : set α) : set α := {x : α | inf_edist x E < ennreal.of_real δ}
+
+/-- The (open) thickening equals the preimage of an open interval under `inf_edist`. -/
+lemma thickening_eq_preimage_inf_edist (δ : ℝ) (E : set α) :
+  thickening δ E = (λ x, inf_edist x E) ⁻¹' (Iio (ennreal.of_real δ)) := rfl
+
+/-- The (open) thickening is an open set. -/
+lemma is_open_thickening {δ : ℝ} {E : set α} : is_open (thickening δ E) :=
+continuous.is_open_preimage continuous_inf_edist _ is_open_Iio
+
+/-- The (open) thickening of the empty set is empty. -/
+@[simp] lemma thickening_empty (δ : ℝ) : thickening δ (∅ : set α) = ∅ :=
+by { unfold thickening, simp only [set_of_false, inf_edist_empty, not_top_lt], }
+
+/-- The (open) thickening `thickening δ E` of a fixed subset `E` is increasing function of the
+thickening radius `δ`. -/
+lemma thickening_mono {δ₁ δ₂ : ℝ} (hle : δ₁ ≤ δ₂) (E : set α) :
+  thickening δ₁ E ⊆ thickening δ₂ E :=
+preimage_mono (Iio_subset_Iio (ennreal.of_real_le_of_real hle))
+
+/-- The (open) thickening `thickening δ E` with a fixed thickening radius `δ` is
+an increasing function of the subset `E`. -/
+lemma thickening_subset_of_subset (δ : ℝ) {E₁ E₂ : set α} (h : E₁ ⊆ E₂) :
+  thickening δ E₁ ⊆ thickening δ E₂ :=
+λ _ hx, lt_of_le_of_lt (inf_edist_le_inf_edist_of_subset h) hx
+
+variables {X : Type u} [metric_space X]
+
+/-- A point in a metric space belongs to the (open) `δ`-thickening of a subset `E` if and only if
+it is at distance less than `δ` from some point of `E`. -/
+lemma mem_thickening_iff {δ : ℝ} (E : set X) (x : X) :
+  x ∈ thickening δ E ↔ (∃ z ∈ E, dist x z < δ) :=
+begin
+  unfold thickening,
+  simp,
```

Comment:
you should avoid non-terminal simps (see https://leanprover-community.github.io/glossary.html#non-terminal-simp)

Reply:
Thank you! This was accidentally left, now corrected.